### PR TITLE
Split up virtool.app.on_shutdown

### PR DESCRIPTION
**Pull request by jakeale**
* Split up into separate tasks in `virtool.shutdown`
* Add tasks to `virtool.app.create_app`

### CODE REVIEW

File: virtool/shutdown.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+async def exit_process_executor(app: aiohttp.web.Application):
+    """
+    Attempt to close the app's `ProcessPoolExecutor`.
+
+    :param app: The application object
+    """
+    try:
+        app["process_executor"].shutdown(wait=True)
+    except KeyError:
+        pass
```
Comment (igboyes) [inline virtool/shutdown.py:59]: Combine to mirror `init_executors`.
```suggestion
async def exit_executors(app: aiohttp.web.Application):
    """
    Attempt to close the `ThreadPoolExecutor` and `ProcessPoolExecutor`.

    :param app: the application object
    """
    try:
        app["executor"].shutdown(wait=True)
    except KeyError:
        pass
    try:
        app["process_executor"].shutdown(wait=True)
    except KeyError:
        pass
```

**Review by igboyes — CHANGES_REQUESTED** [review]

**Comment by jakeale** [comment]
Just took another look at `virtool.startup` and noticed `init_db` and `init_postgres`. Do you want these to be manually shut down and logged as well?

**Comment by igboyes** [comment]
> Just took another look at `virtool.startup` and noticed `init_db` and `init_postgres`. Do you want these to be manually shut down and logged as well?

I'm not sure. Can you do some research on `motor` and `sqlalchemy` to see if this is really necessary?

**Comment by jakeale** [comment]
> > Just took another look at `virtool.startup` and noticed `init_db` and `init_postgres`. Do you want these to be manually shut down and logged as well?
> 
> I'm not sure. Can you do some research on `motor` and `sqlalchemy` to see if this is really necessary?

I could not find the best practices with `motor` but I found this in the `sqlalchemy` docs:

> Valid use cases for calling Engine.dispose() include:
> 
> * When a program wants to release any remaining checked-in connections held by the connection pool and expects to no longer be connected to that database at all for any future operations.
> 
> * When a program uses multiprocessing or fork(), and an Engine object is copied to the child process, Engine.dispose() should be called so that the engine creates brand new database connections local to that fork. Database connections generally do not travel across process boundaries.
> 
> * Within test suites or multitenancy scenarios where many ad-hoc, short-lived Engine objects may be created and disposed.

If there are any open database connections at the time of garbage collection, the `Engine` will keep them alive. Can't immediately think of a case where this could happen.

https://docs.sqlalchemy.org/en/14/core/connections.html?highlight=dispose#engine-disposal

**Review by igboyes — APPROVED** [review]

**Comment by igboyes** [comment]
Thanks for checking into that @jakeale!

**Pull request merged**